tests/go/types_test.go

```diff
@@ -0,0 +1,39 @@
+//go:generate $WIT_BINDGEN_WRPC go --gofmt=false --world types --out-dir bindings/types --package github.com/wrpc/wrpc/tests/go/bindings/types ../wit
+
+package integration_test
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/wrpc/wrpc/tests/go/bindings/types/wrpc_test/integration/get_types"
+)
+
+func TestTypes_Flags(t *testing.T) {
+	t.Run("WriteToIndex", func(t *testing.T) {
+		check := assert.New(t)
+
+		flags := get_types.FeatureFlags{
+			ShowA: true,
+			ShowC: true,
+		}
+		var b bytes.Buffer
+		_, err := flags.WriteToIndex(&b)
+		check.NoError(err)
+		check.Equal([]byte{0b00000101}, b.Bytes())
+	})
+
+	t.Run("ReadFromIndex", func(t *testing.T) {
+		check := assert.New(t)
+
+		var flags get_types.FeatureFlags
+		err := flags.ReadFromIndex(bytes.NewReader([]byte{0b00000101}))
```

Comment:
Could we also test here that reading `0b1000_0000` returns an error? (refs https://github.com/bytecodealliance/wrpc/pull/157/files#r1658495265)